K2Bridge.Tests.UnitTests/Visitors/ElasticSearchDSLVisitorTests.cs

```diff
@@ -17,7 +17,7 @@ namespace UnitTests.K2Bridge.Visitors
     public class ElasticSearchDSLVisitorTests
     {
         [TestCase(ExpectedResult =
-            "let _data = database(\"\").myindex | where (dayOfWeek == 1);\n(_data | limit 0 | as hits)")]
+            "let _data = database(\"\").myindex | where (dayOfWeek == 1);\n(_data | count | as hitsTotal);\n(_data | limit 0 | as hits)")]
```

Comment:
```suggestion
            "let _data = database(\"\").myindex | where (dayOfWeek == 1);\n(_data | count | as hitsTotal);\n(_data | limit 0 | as hits)")]
```


why is hitTotal needed?

Reply:
It has been introduced by @tomconte when he worked on Ranges. I suppose it was related to overlapped ranges not computing the correct addition. Anyway, it also help now to retrieve the total count even there is no aggregation. (before, when you were performing just a count of the all dataset, it was returning only '1').

Reply:
why is it marked as being added now?

Reply:
If you check the code from ElasticSearchDSLVisitor, the **hitsTotal** was previsously added only in case of **aggregations.Count > 0**.  So, in this test, this block was not created.

It's not the case any longer, this hitsTotal block have been moved out and is called even there is no aggregation. This change has been done to catch the correct total count if we don't have any metric/bucket aggregation requested. It fixes the bug 15346. So **hitsTotal** is now added in the test.

Reply:
For reference: 

                // We will need the "true" hits count for some aggregations, e.g. Range
                // And this line must be added even there is no aggregation (default count metric)
                // KQL ==> (_data | count | as hitsTotal);
                queryStringBuilder.Append($"\n(_data | {KustoQLOperators.Count} | as hitsTotal);");

Reply:
@eladiw @AsafMah Don't hesitate to comment this PR this week. I can fix any issue you'll detect. Thanks.